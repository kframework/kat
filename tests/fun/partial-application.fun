let f x = fun [ h : t ] h -> x + h
          |   l         z -> x * z
and apply f = f 4
 in apply (f 3 [ 5 : 6 : .Vals ])

// apply (f 3 [ 5 : 6 : .Vals ])
// f 3 [ 5 : 6 : .Vals ] 4
// 12

// Note if non-linear patterns aren't taken care of with the partial application properly, you get
// 7
