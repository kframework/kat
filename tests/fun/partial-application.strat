f6aec87cfdc9217b5d5c5fe8a4bc9fc6 orig
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx2 exec
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx3 compile
