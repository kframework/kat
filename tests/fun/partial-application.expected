
Strategy: orig
================================================================================
<FUN>
  <k>
    12
  </k>
  <env>
    .Map
  </env>
  <envs>
    .List
  </envs>
</FUN>

Strategy: exec
================================================================================
<kat-FUN>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <FUN>
      <k>
        12
      </k>
      <env>
        .Map
      </env>
      <envs>
        .List
      </envs>
    </FUN>
  </harness>
</kat-FUN>

Strategy: compile
================================================================================
<kat-FUN>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Rules , < <FUN>
        <k>
          let ( f x = fun [ h : t ] h -> x + h | l z -> x * z | .Cases ) and apply f = f 4 and .Bindings in apply ( f 3 ) [ 5 : 6 : .Exps ]
        </k>
        <env>
          .Map
        </env>
        <envs>
          .List
        </envs>
      </FUN> --> <FUN>
        <k>
          12
        </k>
        <env>
          .Map
        </env>
        <envs>
          .List
        </envs>
      </FUN> requires #True >
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <FUN>
      <k>
        12
      </k>
      <env>
        .Map
      </env>
      <envs>
        .List
      </envs>
    </FUN>
  </harness>
</kat-FUN>
